editor/src/components/canvas/canvas-strategies/strategies/reparent-strategy-helpers.ts

```diff
@@ -407,19 +407,27 @@ export function getReparentTargetUnified(
 
   // first try to find a flex element insertion area
   for (const flexElementPath of flexElementsUnderPoint) {
-    const targets: Array<CanvasRectangle> = drawTargetRectanglesForChildrenOfElement(
-      metadata,
-      flexElementPath,
-      'padded-edge',
-      canvasScale,
-    )
+    const targets: Array<{ rect: CanvasRectangle; insertionIndex: number }> =
```

Comment:
instead of using the array index, I explicitly pass around the original index of the element / the index we intend for the inserted element. this makes the code easier to debug, it also means a follow-up PR can exclude elements that do not participate in the layout